src/Z3/Base.hs

```diff
@@ -3512,7 +3515,10 @@ using 'marshal'. Worst case scenario, write the marshalling code yourself.
 -- withIntegral x f = f (fromIntegral x)
 
 withContext :: Context -> (Ptr Z3_context -> IO r) -> IO r
-withContext c = withForeignPtr (unContext c)
+withContext c f = do acquire $ lock c
```

Comment:
Probably better to use  `with :: RLock -> IO a -> IO a `.

Reply:
Yes :+1: